packages/core/src/tools/contextManager.ts

```diff
@@ -1,60 +1,65 @@
-import { computeBytesCount, deepClone, jsonStringify } from './utils'
+import { computeBytesCount, deepClone, jsonStringify, ONE_KIBI_BYTE, throttle } from './utils'
 import type { Context, ContextValue } from './context'
+import type { CustomerDataType } from './heavyCustomerDataWarning'
+import { warnIfCustomerDataLimitReached } from './heavyCustomerDataWarning'
+
+export const CUSTOMER_CONTEXT_BYTES_LIMIT = 5 + ONE_KIBI_BYTE
+export const BYTES_COMPUTATION_THROTTLING_DELAY = 200
 
 export type ContextManager = ReturnType<typeof createContextManager>
 
-export function createContextManager(computeBytesCountImpl = computeBytesCount) {
+export function createContextManager(customerDataType: CustomerDataType, computeBytesCountImpl = computeBytesCount) {
   let context: Context = {}
-  let bytesCountCache: number | undefined
+  let bytesCountCache: number
+
+  const { throttled: computeBytesCountThrottled } = throttle((context: Context) => {
+    bytesCountCache = computeBytesCountImpl(jsonStringify(context)!)
+    warnIfCustomerDataLimitReached(bytesCountCache, customerDataType)
+  }, BYTES_COMPUTATION_THROTTLING_DELAY)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Could you add a bit of reasoning and/or a test for this throttle use?
(same with feature flag context)